mp/src/game/shared/momentum/run/mom_run_safeguards.cpp

```diff
@@ -136,6 +141,7 @@ bool CRunSafeguard::IsDoublePressSafeguarded()
         return true;
     }
 
+    Reset();
```

Comment:
So if I'm reading this correctly, `m_flLastTimePressed` should be set inside the `if (m_bDoublePressSafeguard)` block, so that it is otherwise is reset here.

Reply:
And now that I read this even more, should the first if block absorb the `if (m_bDoublePressSafeguard)` block?

Reply:
You could actually get rid of `m_bDoublePressSafeguard` and just use the time pressed.